www/js/lib/xzdec_wrapper.js

```diff
@@ -20,11 +20,55 @@
  * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
  */
 'use strict';
-define(['q', 'xzdec'], function(Q) {
-    // DEV: xzdec.js emits a global Module variable, which cannot be set in requireJS function line above, though it can be loaded in definition
-    var xzdec = Module;
-    xzdec._init();
-    
+
+// DEV: Put your RequireJS definition in the rqDef array below, and any function exports in the function parenthesis of the define statement
+// We need to do it this way in order to load the wasm or asm versions of xzdec conditionally. Older browsers can only use the asm version
+// because they cannot interpret WebAssembly.
+var rqDefXZ = ['q'];
+
+// Select asm or wasm conditionally
+if ('WebAssembly' in self && !localStorage.getItem(params.keyPrefix + 'boot-with-asm')) {
+    console.debug('Using WASM xz decoder')
+    rqDefXZ.push('xzdec-wasm');
+} else {
+    console.debug('Using ASM xz decoder')
+    rqDefXZ.push('xzdec-asm');
+}
+
+define(rqDefXZ, function(Q) {
+    // DEV: xzdec.js has been compiled with `-s EXPORT_NAME="XZ" -s MODULARIZE=1` to avoid a clash with zstddec.js
+    // Note that we include xzdec-asm or xzdec-wasm above in requireJS definition, but we cannot change the name in the function list
+    // There is no longer any need to load it in index.html
+    // For explanation of loading method below to avoid conflicts, see https://github.com/emscripten-core/emscripten/blob/master/src/settings.js
+
+    /**
+     * @typedef EMSInstance An object type representing an Emscripten instance
+     */
+
+    /**
+     * The XZ Decoder instance
+     * @type EMSInstance
+     */
+     var xzdec;
+     XZ().then(function (instance) {
+         // Instantiate the xzdec object
+         xzdec = instance;
+         if (~rqDefXZ.indexOf('xzdec-wasm')) {
+             localStorage.deleteItem(params.keyPrefix + 'boot-with-asm');
+         }
+     }).catch(function (err) {
+         if (/CompileError.+?WASM/i.test(err.message)) {
```

Comment:
If the error does not match this pattern, it is currently silently ignored : it should be at least logged